functions/sudope.fish

```diff
@@ -1,16 +1,24 @@
-function sudope -d "Quickly toggle sudo prefix"
+function sudope -d "Quickly toggle super user command prefix"
 
   # Save the current command line and cursor position.
   set -l command_buffer (commandline)
   set -l cursor_position (commandline -C)
 
+  # If no custom super user command defaults to sudo
+  if test -z "$sudope_command"
+    set sudope_command 'sudo'
```

Comment:
maybe `set -l` for the fallback here? does anything outside this function need it?

Reply:
Sorry, I'm new to `fish`, but I can read in `help set`:

> If a variable is not explicitly set to be either universal, global or
       local and has never before been defined, the variable will be local to
       the currently executing function. Note that this is different from
       using the -l or --local flag. If one of those flags is used, the
       variable will be local to the most inner currently executing block,
       while without these the variable will be local to the function. If no
       function is executing, the variable will be global.

So I think that all these `-l` flag in this file are not useful. Correct me if I'm wrong, please.

Reply:
using unspecified variable scope introduces opportunities to pollute global state, e.g.:

```fish
set -gx sudope_command

echo $sudope_command # nothing

function thing
  if test -z "$sudope_command"
    set sudope_command sudo
  end
end

thing

echo $sudope_command # "sudo"
```

if you used `set -q` rather than `test -z` to check for a user-defined value, it would have only set the fallback locally. but it also would have failed if `$sudope_command` were set to empty, as above.

i tend to use a pattern that _always_ sets locally, to avoid any chance of this sort of thing. in this case, we could either introduce a brand new variable name, or we can use `set -l` outside a block:

```fish
set -gx sudope_command

echo $sudope_command # nothing

function thing
  test -z "$sudope_command"
    or set -l sudope_command sudo
end

thing

echo $sudope_command # nothing
```

Reply:
Now I understand. Thanks for your explanation and your patience.

Reply:
Fixed in 18d9fd255ea069093d9c8c862767b0db306fbcf4.